lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,23 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion: string | null = null;
+    const prerelease = semver.prerelease(currentValue);
```

Comment:
Why is this logic executed without checking if the current version is a snapshop/pre-release? Additionally, this would only work with semver-compatible versions, possibly even error for incompatible.